05_iigo/index.tex

```diff
@@ -500,4 +500,226 @@ \subsection{Legislative Branch}
+\[
+    \begin{bmatrix}
+        -1 & 1 & -5
+    \end{bmatrix}
+    \begin{bmatrix}
+        x \\
+        y \\
+        1
+    \end{bmatrix}   
+    >= 0  
+\]
+We have now managed to capture any linear equation or inequality with any constant shift. This is where we decided to limit the scope of our rules from a technical aspect.
+The final section of our schema is how to inform an agent whether it should be looking for equality or inequality.
+For this, we package with the rule-matrix an "auxiliary" vector containing a list of comparison codes.
+\begin{table}[h]
+    \centering
+    \begin{tabular}{ll}
+    \hline
+    \multicolumn{1}{|l}{Aux code} & \multicolumn{1}{l|}{Meaning} \\ \hline
+    0                             & = 0                          \\
+    1                             & \textgreater 0               \\
+    2                             & \textgreater{}= 0            \\
+    3                             & != 0                         \\
+    4                             & real                        
+    \end{tabular}
+\end{table}
+\linebreak
+The aux code 4 is reserved for when we want the raw result of the matrix multiplication, which is used sparingly for certain rules.
+We have now arrived at the representation of rules we are using for our project. For the rule above:
+\[
+    \textrm{Rule matrix: }
+    \begin{bmatrix}
+        -1 & 1 & -5
+    \end{bmatrix}
+    \textrm{   Variables to fill:  }
+    \begin{bmatrix}
+        \textrm{Amount of tax agent is asked to pay} \\
+        \textrm{Amount of tax the agent is actually paying} \\
+        1
+    \end{bmatrix}     
+\]
+\[
+    \textrm{   Auxiliary vector: } 
+    \begin{bmatrix}
+        2 
+    \end{bmatrix}   
+\]
+Let us revist our tax rule once more, adding a further layer of complexity.
+\linebreak
+\linebreak
+"You must pay in tax at least half as many resources as you have and if you've been told to pay tax x you must pay at least tax of x."
+In this example there are two separate conditions. We can write both conditions as separate rules.
+Condition 1:
+\[
+    \textrm{Rule matrix: }
+    \begin{bmatrix}
+        0 & 2 & -1 & 0
+    \end{bmatrix}
+    \textrm{   Variables to fill:  }
+    \begin{bmatrix}
+        \textrm{Amount of tax agent is asked to pay} \\
+        \textrm{Amount of tax the agent is actually paying} \\
+        \textrm{Amount of resources agent has} \\
+        1
+    \end{bmatrix}     
+\]
+\[
+    \textrm{   Auxiliary vector: } 
+    \begin{bmatrix}
+        2 
+    \end{bmatrix}   
+\]
+\linebreak
+Condition 2:
+\[
+    \textrm{Rule matrix: }
+    \begin{bmatrix}
+        -1 & 1 & 0 & 0
+    \end{bmatrix}
+    \textrm{   Variables to fill:  }
+    \begin{bmatrix}
+        \textrm{Amount of tax agent is asked to pay} \\
+        \textrm{Amount of tax the agent is actually paying} \\
+        \textrm{Amount of resources agent has} \\
+        1
+    \end{bmatrix}     
+\]
+\[
+    \textrm{   Auxiliary vector: } 
```

Comment:
wait.. isn't this a scalar lol

Reply:
All scalars are a 1-d vector, very smol region of space.